PowerShellTools/ServiceManagement/ChannelFactoryMaker.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Net.Security;
+using System.ServiceModel;
+using PowerShellTools.Common;
+
+namespace PowerShellTools.ServiceManagement
+{
+    internal static class ChannelFactoryMaker<ServiceType> where ServiceType : class
+    {
+        private static ChannelFactory<ServiceType> _channelFactory;
+        private static object _syncObject = new object();
+
+        public static string EndPointAddress { get; set; }
+
+        public static ChannelFactory<ServiceType> CreateChannelFactory()

```

Comment:
Check out the latest commit I just made…

From: Eric Deng [mailto:notifications@github.com]
Sent: Monday, January 26, 2015 11:52 AM
To: Microsoft/poshtools
Cc: Huanhuan Sun
Subject: Re: [poshtools] Add exception handling mechanism for process and service host. (#60)


In PowerShellTools/ServiceManagement/ChannelFactoryMaker.cs<https://github.com/Microsoft/poshtools/pull/60#discussion_r23558882>:

> @@ -0,0 +1,31 @@

> +﻿using System;

> +using System.Net.Security;

> +using System.ServiceModel;

> +using PowerShellTools.Common;

> +

> +namespace PowerShellTools.ServiceManagement

> +{

> +    internal static class ChannelFactoryMaker<ServiceType> where ServiceType : class

> +    {

> +        private static ChannelFactory<ServiceType> _channelFactory;

> +        private static object _syncObject = new object();

> +

> +        public static string EndPointAddress { get; set; }

> +

> +        public static ChannelFactory<ServiceType> CreateChannelFactory()

Looks like this factory method is requiring the endpoint address, why not taking it as parameter? Coz consumer of this factory method does not have to know they need initialize a static field before suing this static method....

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/poshtools/pull/60/files#r23558882>.